src/mpp/Dec.hpp

```diff
@@ -688,16 +699,39 @@ struct JumpsBuilder {
 	}
 };
 
+
 template <class PATH, class BUF, class... T>
 bool
-decode_impl(BUF& buf, T... t)
+decode_jump(BUF& buf, T... t)
 {
 	static_assert(path_item_type(PATH::last()) != PIT_BAD);
 	static constexpr auto jumps = JumpsBuilder<PATH, BUF, T...>::build();
 	uint8_t tag = buf.template get<uint8_t>();
 	return jumps.data[tag](buf, t...);
 }
 
+template <class PATH, class BUF, class... T>
+bool
+decode_impl(BUF& buf, T... t)
+{
+	static_assert(path_item_type(PATH::last()) != PIT_BAD);
+	if constexpr (path_item_type(PATH::last()) != PIT_DYN_SKIP &&
+		      path_item_type(PATH::last()) != PIT_DYN_RAW) {
+		auto&& wrapped_dst = path_resolve(PATH{}, t...);
+		using wrapped_dst_t = std::remove_reference_t<decltype(wrapped_dst)>;
+		if constexpr (is_raw_decoded_v<wrapped_dst_t, BUF>) {
+			using RAW_PATH = path_push_t<PATH, PIT_DYN_RAW>;
+			using RAW_SKIP_PATH = path_push_t<RAW_PATH, PIT_DYN_SKIP>;
+			BUF begin = buf;
+			return decode_impl<RAW_SKIP_PATH>(buf, t..., begin, size_t(1));
```

Comment:
Could you please move this code into a separate `decode_` helper and elaborate what's going on? I don't understand what the `size_t(1)` and `RAW_SKIP_PATH` are needed for.

Reply:
`size_t(1)` is for `PIT_DYN_SKIP` - to skip one object.